validator-network/src/network_impl.rs

```diff
@@ -195,9 +195,10 @@ where
                 if let Some(validator_key) =
                     self.validator_keys.read().get(usize::from(validator_id))
                 {
-                    self.validator_peer_id_cache
-                        .write()
-                        .remove(validator_key.compressed());
+                    self.validator_peer_id_cache.write().insert(
+                        validator_key.compressed().clone(),
+                        CacheState::Error(NetworkError::UnknownValidator(validator_id)),
+                    );
```

Comment:
This should probably only be done if this invalidates the cache entry that led to the error. Would that be possible?

Reply:
Not sure if I'm following you, this is trying to invalidate the cache entry that led to the error

Reply:
I'm imagining the following scenario:

The current cache entry of validator 1 is invalid. Two requests to that are currently in flight. The first one fails and clears the cache entry. Another request is sent, causing the cache to be updated to the correct value. The second initial request fails, and now clears the cache value even though it's already been updated.

Reply:
Yeah, thanks! Updated PR to incorporate the change we discussed